### PR TITLE
UCT/MM: Use unix socket for wakeup instead of on-demand progress.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4212/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@alinask pls review

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4217/ for details (Mellanox internal link).

File: src/uct/sm/mm/mm_ep.h [hunk 1]
```diff
@@ -30,13 +30,11 @@ struct uct_mm_ep {
...
-     * while this is not necessary for mmap() call, it is critical for XPMEM */
-    socklen_t            cached_signal_addrlen;   /* cached address length of signaling socket */
-    struct sockaddr_un   cached_signal_sockaddr;  /* cached address of signaling socket */
-
-    uct_worker_cb_id_t   slow_cb_id; /* Slow-path callback */
+    /* Used for signaling remote side wakeup */
+    struct {
+        struct sockaddr_un  sockaddr;  /* address of signaling socket */
+        socklen_t           addrlen;   /* address length of signaling socket */
+    } signal;
```
Comment (alinask) [inline src/uct/sm/mm/mm_ep.h]: can use ucs_sock_addr_t after client-server api goes in i guess.

File: src/uct/sm/mm/mm_iface.c [hunk 2]
```diff
@@ -256,6 +258,37 @@ unsigned uct_mm_iface_progress(void *arg)
...
+{
+    *fd_p = ucs_derived_of(tl_iface, uct_mm_iface_t)->signal_fd;
+    return UCS_OK;
+}
+
+static ucs_status_t uct_mm_iface_event_fd_arm(uct_iface_h tl_iface,
+                                              unsigned events)
+{
+    uct_mm_iface_t *iface = ucs_derived_of(tl_iface, uct_mm_iface_t);
+    char dummy[32]; /* pop multiple signals at once */
```
Comment (alinask) [inline src/uct/sm/mm/mm_iface.c]: can you please explain why it's 32? i dont see where this value is used

Comment (yosefe) [inline src/uct/sm/mm/mm_iface.c]: it's used below when doing sizeof(dummy)

File: src/uct/sm/mm/mm_ep.h (hunk 1 above)
Comment (yosefe) [inline src/uct/sm/mm/mm_ep.h]: yep

File: src/uct/sm/mm/mm_iface.c (hunk 2 above)
Comment (alinask) [inline src/uct/sm/mm/mm_iface.c]: yes, but i meant why 32?

File: src/uct/sm/mm/mm_ep.c [hunk 3]
```diff
@@ -275,9 +208,10 @@ static inline void uct_mm_ep_update_cached_tail(uct_mm_ep_t *ep)
  * is_short = 0 - perform AM bcopy sending
  */
 static UCS_F_ALWAYS_INLINE ssize_t
-uct_mm_ep_am_common_send(const unsigned is_short, uct_mm_ep_t *ep, uct_mm_iface_t *iface,
```
Comment (alinask) [inline src/uct/sm/mm/mm_ep.c]: why is const removed?

File: src/uct/sm/mm/mm_iface.c (hunk 2 above)
Comment (yosefe) [inline src/uct/sm/mm/mm_iface.c]: just some reasonable value of how many events to read at once
not to large to fit on the stack, and large enough to reduce overhead of reading events

File: src/uct/sm/mm/mm_ep.c (hunk 3 above)
Comment (yosefe) [inline src/uct/sm/mm/mm_ep.c]: because it's meaningless for value arguments

File: src/uct/sm/mm/mm_iface.c (hunk 2 above)
Comment (shamisp) [inline src/uct/sm/mm/mm_iface.c]: then define it with MAX_EVENTS

File: src/uct/sm/mm/mm_ep.c [hunk 4]
```diff
@@ -353,6 +287,10 @@ uct_mm_ep_am_common_send(const unsigned is_short, uct_mm_ep_t *ep, uct_mm_iface_
         elem->flags &= ~UCT_MM_FIFO_ELEM_FLAG_OWNER;
     }
 
+    if (flags & UCT_AM_FLAG_SIGNALED) {
```
Comment (shamisp) [inline src/uct/sm/mm/mm_ep.c]: unlikely ? anyways signaling is already expensive ...

**Comment by shamisp** [comment]
👍

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4239/ for details (Mellanox internal link).

**Pull request merged**